src/test/java/com/ms/zookeeper/TestZkUserAuthorizer.java

```diff
@@ -0,0 +1,47 @@
+package com.ms.zookeeper;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import org.junit.Test;
+
+import com.ms.zookeeper.authz.ZkUserAuthorizer;
+
+public class TestZkUserAuthorizer {
+    @Test
+    public void testUserAuth() {
+        ZkUserAuthorizer auth = new ZkUserAuthorizer();
+        assertTrue(auth.authorize("username", "user://username"));
+        assertFalse(auth.authorize("username", "user://otherusername"));
+    }
+
+    @Test
+    public void testUserAuthWithRealms() {
+        System.setProperty("zookeeper.validKerberosRealms", "realm,realm2");
+        ZkUserAuthorizer auth = new ZkUserAuthorizer();
+        assertTrue(auth.authorize("username@REALM", "user://username"));
```

Comment:
back to previous q - do we have to strip realm?

Reply:
We do strip the realm internally. I don't want to fork this functionality so it is easier to maintain. 
This is an optional feature. If validKerberosRealms are not configure than strip won't work